base_wamas/models/wamas_document_field.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2023 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import date, datetime
+
+from dateutil.parser import parse
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import safe_eval
+
+FIELD_TYPES = [
+    ("str", "String"),
+    ("int", "Integer"),
+    ("float", "Float"),
+    ("date", "Date"),
+    ("datetime", "Datetime"),
+    ("bool", "Boolean"),
+]
+
+
+def is_date(string, fuzzy=False):
+    """
+    Return whether the string can be interpreted as a date.
+
+    :param string: str, string to check for date
+    :param fuzzy: bool, ignore unknown tokens in string if True
+    """
+    try:
+        parse(string, fuzzy=fuzzy)
+        return True
+
+    except ValueError:
+        return False
+
+
+class WamasDocumentField(models.Model):
+    _name = "wamas.document.field"
+    _description = "WAMAS Document Field"
+
+    name = fields.Char(required=True)
+    code = fields.Char(required=True)
+    ttype = fields.Selection(FIELD_TYPES, "Type", required=True)
+    len_field = fields.Integer("Length", required=True)
+    decimal_place = fields.Integer()
+    sequence = fields.Integer(required=True, default=10)
+    element_id = fields.Many2one("wamas.document.element", "Element")
+
+    def get_value_from_string(self, value):
+        self.ensure_one()
+
+        res = value
+
+        if self.ttype == "str":
+            res = self._get_string(value)
+        elif self.ttype == "int":
+            res = self._get_int(value)
+        elif self.ttype == "float":
+            res = self._get_float(value)
+        elif self.ttype == "date":
+            res = self._get_date(value)
+        elif self.ttype == "datetime":
+            res = self._get_datetime(value)
+        elif self.ttype == "bool":
+            res = self._get_bool(value)
```

Comment:
```suggestion
        getters = dict(
            str=self._get_string,
            int=self._get_int
            [....]
        )
        return getters[self.ttype](value)
```
Those methods should be renamed like... `_get_value_from_string__[$ttype]`

An alternative to the dict of getters is:

```
handler = getattr(self, f"_get_value_from_string__{self.ttype}`)
return handler(value)
```